src/libraries/Common/src/System/IO/Archiving.Utils.Windows.cs

```diff
@@ -9,19 +9,38 @@ internal static partial class ArchivingUtils
     {
         internal static string SanitizeEntryFilePath(string entryPath)
         {
-            StringBuilder builder = new StringBuilder(entryPath);
+            // Find the first illegal character in the entry path.
             for (int i = 0; i < entryPath.Length; i++)
             {
-                if (((int)builder[i] >= 0 && (int)builder[i] < 32) ||
-                   builder[i] == '?' || builder[i] == ':' ||
-                   builder[i] == '*' || builder[i] == '"' ||
-                   builder[i] == '<' || builder[i] == '>' ||
-                   builder[i] == '|')
+                switch (entryPath[i])
                 {
-                    builder[i] = '_';
+                    // We found at least one character that needs to be replaced.
+                    case < (char)32 or '?' or ':' or '*' or '"' or '<' or '>' or '|':
```

Comment:
Should I get used to switch statements with only one case?

Reply:
@stephentoub Is there a performance advantage vs. `if (entryPath[i] is < (char)32 or ...` or is it just your preferred style?
[SharpLab comparison](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT1sAGEgWQAoBKGuhgczYBTGABcATgE8AChDEALEgF4SMAK4AbTQG5+AhgDMA9hJKdYYklBUkWO6yQA8JUZNnyFAOgAyogOaKDlDIyLy0+AZR1obmbtJyigDaUAC61gDOzuYAxgoQEtxEmCSmJADkAPzlpWblIDVl5QBUjXUARG0VTl3lAHy9qOXh+tEG6ADsJO0Q7XqRYwwAvgSjYxkA7lBieXHiCZ4pqSMLiwI5EBnC2Zx5BUUlTdW1FQ0vLb2d7z3vA+9DIDWZwYk2mwDmQKiK1OUMh9FB8Q8QVG0KWQA)